contracts/multiswap-contracts/CCTPFundManager.sol

```diff
@@ -0,0 +1,249 @@
+     * It checks if the sender is equal to the `fiberRouter` address.
+     * @notice Ensure that `fiberRouter` is set before using this modifier.
+     */
+    modifier onlyRouter() {
+        require(msg.sender == fiberRouter, "FM: Only fiberRouter method");
+        _;
+    }
+
+    /**
+     * @dev Contract constructor that initializes the EIP-712 domain with the specified NAME, VERSION.
+     * @notice This constructor is called only once during the deployment of the contract.
+     */
+    constructor() EIP712(NAME, VERSION) {
+        // bytes memory initData = IFerrumDeployer(msg.sender).initData();
+    }
+
+    /**
+     @dev sets the fiberRouter
+     @param _fiberRouter is the FiberRouter address
+     */
+    function setRouter(address _fiberRouter) external onlyOwner {
+        require(_fiberRouter != address(0), "FM: fiberRouter requried");
+        fiberRouter = _fiberRouter;
+    }
+
+    /**
+     @dev sets the signer
+     @param _signer is the address that generate signatures
+     */
+    function addSigner(address _signer) external onlyOwner {
+        require(_signer != address(0), "Bad signer");
+        signers[_signer] = true;
+    }
+
+    /**
+     @dev removes the signer
+     @param _signer is the address that generate signatures
+     */
+    function removeSigner(address _signer) external onlyOwner {
+        require(_signer != address(0), "Bad signer");
+        delete signers[_signer];
+    }
+
+    /**
+     * @dev Initiates a signed token withdrawal, exclusive to the router
+     * @notice Ensure valid parameters and router setup
+     * @param token The token to withdraw
+     * @param payee Address for where to send the tokens to
+     * @param amount The amount
+     * @param salt The salt for unique tx
+     * @param expiry The expiration time for the signature
+     * @param signature The multisig validator signature
+     * @return The actual amount of tokens withdrawn
+     */
+    function withdrawSigned(
+        address token,
+        address payee,
+        uint256 amount,
+        bytes32 salt,
+        uint256 expiry,
+        bytes memory signature
+    ) external onlyRouter returns (uint256) {
+        require(token != address(0), "FM: bad token");
+        require(payee != address(0), "FM: bad payee");
+        require(salt != 0, "FM: bad salt");
+        require(amount != 0, "FM: bad amount");
+        require(block.timestamp < expiry, "FM: signature timed out");
+        require(expiry < block.timestamp + WEEK, "FM: expiry too far");
+        bytes32 message =  keccak256(
+                abi.encode(WITHDRAW_SIGNED_METHOD, token, payee, amount, salt, expiry)
+            );
+        address _signer = signerUnique(message, signature);
+
+        require(signers[_signer], "FM: Invalid signer");
+        require(!usedSalt[salt], "FM: salt already used");
+        usedSalt[salt] = true;
+        // sync inventory of token
+        TokenReceivable.sync(token);
```

Comment:
Why do we need a sync here? Do we send token to this contract within the same transaction calling `withdrawSigned`? It will only cost gas.